tests/test_config_parser.py

```diff
@@ -303,6 +307,33 @@ def test_substring_reference(self, config, expected):
         parser = ConfigParser(config=config)
         self.assertEqual(parser.get_parsed_content("total"), expected)
 
+    @parameterized.expand([TEST_CASE_DUPLICATED_KEY])
+    @mock.patch.dict(os.environ, {"MONAI_FAIL_ON_DUPLICATE_CONFIG": "1"})
+    def test_parse_json_raise(self, config_string, _, __):
```

Comment:
Can we combine these 2 test functions `test_parse_json_raise` and `test_parse_json_warn`?
Seems most of the logic are similar.

Thanks.

Reply:
I think it's gonna reduce the readability a lot. 
1. We use `@mock.patch.dict(os.environ, {"MONAI_FAIL_ON_DUPLICATE_CONFIG": "1"})` which as an annotation nicely handles changing and restoring os.environ
2. The two scenarios differ by:
- the context manager in which the tested method is applied in
- how the expected reaction is verified
- additional assertions

Introducing all this as parameters would make the main test method abstract and diluted.

At the same time - I've introduced this functionality for yaml without introducing new test methods, since this nicely fits as a parameter

Reply:
OK, that makes sense. thanks.